src/android/FileUtils.java

```diff
@@ -73,8 +73,11 @@ Licensed to the Apache Software Foundation (ASF) under one
      * Permission callback codes
      */
 
-    public static final int READ_PERM = 0;
-    public static final int WRITE_PERM = 1;
+    public static final int PERM_FILE = 0;
+    public static final int WRITE_ACTION = 1;
+    public static final int PERM_DIR = 2;
+    public static final int WRITE = 3;
+    public static final int READ = 4;
```

Comment:
WRITE_ACTION corresponds to the handler for the write api call. READ and WRITE corresponds to the permission being asked for.